scripts/info.py

```diff
@@ -0,0 +1,118 @@
+import argparse
+import os
+from pathlib import Path
+from typing import List
+from datasets import load_dataset_builder
+import sys
+
+INFO_KEYS = ['citation', 'description', 'homepage', 'license', 'version', 'acknowledgements']
+
+def get_all_datasets() -> List[str]:
+    # Collect all unique dataset names from the scripts directory
+    all_datasets = set()
+    # only consider directories in the scripts directory
+    for file in os.listdir('scripts'):
+        if os.path.isdir(os.path.join('scripts', file)):
+            all_datasets.add(file)
+    return sorted(list(all_datasets))
+
+def get_info(dataset: str, info_keys: List[str] = ['citation']) -> dict:
+    try:
+        # Construct path to the dataset script
+        script_path = Path('scripts') / f"{dataset}/{dataset}.py"
+        if not script_path.exists():
+            raise FileNotFoundError(f"Dataset script not found: {script_path}")
+            
+        # Load the dataset builder from the local script
+        sys.path.insert(0, str(script_path.parent))
+        builder = load_dataset_builder(str(script_path), trust_remote_code=True)
+        sys.path.pop(0)
+        
+        info = builder.info
+        
+        # Get the requested information
+        dataset_info = {}
+        if "acknowledgements" in info_keys:
+            if info_keys == INFO_KEYS:
+                info_keys.remove("acknowledgements")
+            else:
+                # Remove all % from line starts 
+                if info.citation.startswith(r"% % ACKNOWLEDGEMENTS") or info.citation.startswith(r"% ACKNOWLEDGEMENTS"):
+                    dataset_info["acknowledgements"] = "\n".join([line.lstrip("% ") for line in info.citation.split("% CITATION\n")[0].split("\n")])
+                else:
+                    dataset_info["acknowledgements"] = "% ACKNOWLEDGEMENTS\nNot available"
+        for key in info_keys:
+            if key == "acknowledgements":
+                continue
+            value = getattr(info, key, "Not available")
+            dataset_info[key] = value if value else "Not available"
+        return dataset_info
+    except Exception as e:
```

Comment:
Could we avoid long try and except clauses? What is the problem that we may face and we want to catch?

Reply:
I have refactored to make the process clearer and only try/except the data loading which is what I was initially worried about. Thanks for catching this.